### PR TITLE
Raise request to merge from rick-test-dev into dev branch

### PR DESCRIPTION
**Pull request by rickchen918**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -11,6 +11,28 @@ variable "ha_distribution" {
   }
 }
 
+variable "standby_instance_state" {
+  type            = string
+  description     = "Standby instance state definition"
+  default         = "Running"
+  validation {
+    condition     = contains (["Running", "Stopped", "Hibernated"], var.standby_instance_state)
+    error_message = "Valid values for var: ha_distribution are (Running, Stopped and Hibernated)."
```
Comment (sshussai) [inline variables.tf:20]: this should be `var: standby_instance_state` not `var: ha_distribution`.

File: region-build/variables.tf [hunk 2]
```diff
@@ -11,6 +11,29 @@ variable "ha_distribution" {
   }
 }
 
+variable "standby_instance_state" {
+  type        = string
+  description = "Standby instance state definition"
```
Comment (sshussai) [inline region-build/variables.tf:16]: should we add a default value here too?

File: region-build/variables.tf [hunk 3]
```diff
@@ -11,6 +11,29 @@ variable "ha_distribution" {
...
+variable "copilot_ha_enabled" {
+  type        = bool
+  description = "AWS autoscale group functions suspending for controller copilot"
+  default     = true
+}
+
+variable "suspended_processes" {
+  type        = list
+  description = "ASG process types suspending"
+  default     = ["Launch","Terminate","HealthCheck","ReplaceUnhealthy"]
```
Comment (sshussai) [inline region-build/variables.tf:34]: does this mean we're suspending all those processes in the ASG by default?

Comment (rickchen918) [reply region-build/variables.tf:34]: udpated the fix

File: variables.tf (hunk 1 above)
Comment (rickchen918) [reply variables.tf]: update the fix

File: region-build/variables.tf (hunk 2 above)
Comment (rickchen918) [reply region-build/variables.tf:16]: default value is specified on upper layer , becuase customer may give ture or false 

variable "standby_instance_state" {
  type            = string
  description     = "Standby instance state definition"
  default         = "Running"
  validation {
    condition     = contains (["Running", "Stopped", "Hibernated"], var.standby_instance_state)
    error_message = "Valid values for var: standby_instance_state are (Running, Stopped and Hibernated)."
  }
}

File: region-build/variables.tf (hunk 3 above)
Comment (rlee-aviatrix) [reply region-build/variables.tf:34]: Just to clarify, those are the processes that are suspended if `var.controller_ha_enabled` is `false` (it's `true` by default).

```
resource "aws_autoscaling_group" "avtx_ctrl" {
  name                      = "avtx_controller"
  max_size                  = 1
  min_size                  = 0
  health_check_grace_period = 300
  health_check_type         = "ELB"
  desired_capacity          = 1
  force_delete              = true
  suspended_processes       = var.controller_ha_enabled ? null : var.suspended_processes
```

**Comment by rickchen918** [comment]
the chnage is not updated in PR .. I will re-raise the PR

**Pull request closed without merging**